beacon-chain/core/state/transition.go

```diff
@@ -272,10 +273,18 @@ func ProcessSlots(ctx context.Context, state iface.BeaconState, slot types.Slot)
 			return nil, errors.Wrap(err, "could not process slot")
 		}
 		if CanProcessEpoch(state) {
-			state, err = ProcessEpochPrecompute(ctx, state)
-			if err != nil {
-				traceutil.AnnotateError(span, err)
-				return nil, errors.Wrap(err, "could not process epoch with optimizations")
+			if state.Version() == version.Altair {
```

Comment:
Review this. Calls different `ProcessEpoch` based on state versioning

Reply:
Can you also do the switch via version here too for phase0,altair, default, etc,